main.cpp

```diff
@@ -76,17 +77,27 @@ void findFirstShapeMatchingPredicate(const Collection& collection,
     }
 }
 
+constexpr int fibo(int number)
+{
+    if (number == 1 || number == 0)
+    {
+        return number;
+    }
+    return (fibo(number-1) + fibo(number -2));
+}
+
 int main()
 {
-    Collection shapes;
-    shapes.push_back(new Circle(2.0));
-    shapes.push_back(new Circle(3.0));
-    shapes.push_back(NULL);
-    shapes.push_back(new Circle(4.0));
-    shapes.push_back(new Rectangle(10.0, 5.0));
-    shapes.push_back(new Square(3.0));
-    shapes.push_back(new Circle(4.0));
+    //static_assert(M_PI == 3.14, "wrong"); //zadanie 1
 
+    int n = fibo(45);
+    std::cout<< n;
+    alignas(2) Circle a(1);    
+    std::cout<< "\n" << sizeof(a) << std::endl;
+    static_assert(alignof(a) == 2);
+
+    Collection shapes = std::initializer_list<std::shared_ptr<Shape>>{new Circle(2.0), new Circle(3.0), nullptr, new Circle(4.0), new Rectangle(10.0, 5.0), new Square(3.0), new Circle(4.0)};
```

Comment:
lepiej zrobic tak 
```cpp
Collection shapes = {
    std::make_shared<Circle>(2.0),
    ...
};
```

Reply:
Corrected, but i had problems when i tried created shared pointer by std::make_shared<Shape>(nullptr), because it must point on something, that's why i changed one line in this initialize list:

    Collection shapes = {
        std::make_shared<Circle>(2.0),
        std::make_shared<Circle>(3.0),
        std::shared_ptr<Shape>(nullptr),
        std::make_shared<Circle>(4.0),
        std::make_shared<Rectangle>(10.0, 5.0),
        std::make_shared<Square>(3.0),
        std::make_shared<Circle>(4.0)
                        };

But You had right, code in that way is more readable:)